packages/ember-data/lib/system/model/model.js

```diff
@@ -375,7 +374,7 @@ var Model = Ember.Object.extend(Ember.Evented, {
   */
   toJSON: function(options) {
     // container is for lazy transform lookups
-    var serializer = JSONSerializer.create({ container: this.container });
+    var serializer = this.store.serializerFor('-default');
```

Comment:
Would it be possible to do something like `store.serializerFor(this.constructor.typeKey)` instead? To get the record-specific serializer when one is available.

We've had this work-around for a long time (code not aligned with recent ED, but it shows the main idea):
```
toJSON: function(options) {
  var serializer = this.store.serializerFor(this.constructor.typeKey);
  return serializer.serialize(this, options);
}
```